Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp

```diff
@@ -286,7 +286,7 @@ std::optional<BoundVector> estimateTrackParamsFromSeed(
   ActsScalar pathz = spGlobalPositions[0].dot(bField) / bField.norm();
   // The estimated time (use path length along magnetic field only if it's not
   // zero)
-  if (pathz != 0) {
+  if (pathz != 0 && vz != 0) {
     params[eBoundTime] = pathz / vz;
```

Comment:
Shouldn't we use 

`vLongitudinal = vVector.dot(bField) / bField.norm() / vVector.norm()`, 

where vVector is the vector of the velocity,

instead of `vz`?

Reply:
Is that not just the velocity fraction along z?
And does that not have the same problem, where it gives `0` if the momentum is perpendicular to the B field? I don't think we don't need `vx` and `vy` explicitly, so should we really calculate them?

Reply:
Sorry, I forgot to mention that this was not in relation with your change!

I was just wondering because pathz is calculated as a projection on the B field a few lines above (implicitely assuming that the B field points in z direction I guess?).  To get the correct time for all orientations of the B field, I think we also need to calculate the projection of the velocity on the latter.

Sorry to bring this up here - I know it is not you who wrote the code, but I was wondering about this :)

Reply:
Your change seems fine, although I find it worrysome that sometimes `pathz !=0` when `vz==0`; a 0 velocity should imply a 0 path right?